Failure to find ffmpeg/libavcodec header files during compilation

### **Description**

Linux build script fails to find libavcodec header files on Fedora 43 with ffmpeg-devel from rpmfusion.

Component: lib_ccx
File: src/lib_ccx/hardsubx.c and src/lib_ccx/hardsubx.h
### **Problem**

Compilation of ccextractor fails on systems where ffmpeg header files exist under /usr/include/ffmpeg. The header files are installed, they're just in a weird location and the environment variables aren't fixing it.

Compilation command used:
```
cd linux
export ENABLE_FFMPEG=1 FFMPEG_DLL_PATH='/usr/lib64' FFMPEG_INCLUDE_DIR='/usr/include/ffmpeg' FFMPEG_PKG_CONFIG_PATH='/usr/lib64/pkgconfig' RUST_BACKTRACE=full
./build -hardsubx -system-libs
```

Error message:
```
Running `/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/build/ccx_rust-e0b85611ce9f5505/build-script-build`
error: failed to run custom build command for `ccx_rust v0.1.0 (/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust)`
note: To improve backtraces for build dependencies, set the CARGO_PROFILE_RELEASE_BUILD_OVERRIDE_DEBUG=true environment variable to enable debug information generation.

Caused by:
  process didn't exit successfully: `/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/build/ccx_rust-e0b85611ce9f5505/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=TARGET
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS
  cargo:rerun-if-changed=wrapper.h

  --- stderr
  ./../lib_ccx/hardsubx.h:9:10: fatal error: 'libavcodec/avcodec.h' file not found

  thread 'main' (2044) panicked at build.rs:143:10:
  Unable to generate bindings: ClangDiagnostic("./../lib_ccx/hardsubx.h:9:10: fatal error: 'libavcodec/avcodec.h' file not found\n")
```

### **Proposed Fix**
I haven't figured out how to fix this.

### **Impact**
This issue causes compilation to fail.

This has been an issue for years and there's still a comment in the header and c files that say "// TODO: Correct FFMpeg integration", but there was a similar issue a while back that got closed as "not planned". I have been attempting to fix this for a while but I have no idea what I'm doing, is there at least a good work-around if it's not going to be fixed?

@cfsmp3 so it looks like it worked, and I've gotten slightly farther into the build process. However, a new problem has appeared and I have no idea what it means:
```
Compiling ccx_rust v0.1.0 (/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust)
     Running `/usr/bin/rustc --crate-name build_script_build --edition=2018 build.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debug-assertions=off --cfg 'feature="hardsubx_ocr"' --cfg 'feature="leptonica-sys"' --cfg 'feature="rsmpeg"' --cfg 'feature="tesseract-sys"' --check-cfg 'cfg(docsrs,test)' --check-cfg 'cfg(feature, values("enable_ffmpeg", "hardsubx_ocr", "leptonica-sys", "rsmpeg", "sanity_check", "tesseract-sys", "with_libcurl", "wtv_debug"))' -C metadata=cce8eec4c5886004 -C extra-filename=-e0b85611ce9f5505 --out-dir /builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/build/ccx_rust-e0b85611ce9f5505 -C strip=debuginfo -L dependency=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps --extern bindgen=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libbindgen-079ee1562ef80e50.rlib --extern pkg_config=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libpkg_config-077987094adac708.rlib -Copt-level=3 -Cdebuginfo=2 -Ccodegen-units=1 -Cstrip=none -Cforce-frame-pointers=yes --cap-lints=warn`
     Running `/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/build/ccx_rust-e0b85611ce9f5505/build-script-build`
     Running `/usr/bin/rustc --crate-name ccx_rust --edition=2018 src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type staticlib --emit=dep-info,link -C opt-level=3 -C embed-bitcode=no --cfg 'feature="hardsubx_ocr"' --cfg 'feature="leptonica-sys"' --cfg 'feature="rsmpeg"' --cfg 'feature="tesseract-sys"' --check-cfg 'cfg(docsrs,test)' --check-cfg 'cfg(feature, values("enable_ffmpeg", "hardsubx_ocr", "leptonica-sys", "rsmpeg", "sanity_check", "tesseract-sys", "with_libcurl", "wtv_debug"))' -C metadata=52cada234fc33f24 -C extra-filename=-6d45b476bf719ea7 --out-dir /builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps -C strip=debuginfo -L dependency=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps --extern cfg_if=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libcfg_if-9d2dbc57d8792789.rlib --extern clap=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libclap-e5fc0d6cf5f0b011.rlib --extern crc32fast=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libcrc32fast-fbcf64dd4a2f897e.rlib --extern encoding_rs=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libencoding_rs-233b529cd9a1c6e3.rlib --extern env_logger=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libenv_logger-472fef667cb831cb.rlib --extern leptonica_sys=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libleptonica_sys-a9696d6acc70da1d.rlib --extern lib_ccxr=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/liblib_ccxr-4e3e582127ca6239.rlib --extern log=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/liblog-7bc3130406e50940.rlib --extern num_integer=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libnum_integer-69083584aceac81d.rlib --extern palette=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libpalette-b61653662ba8d509.rlib --extern rsmpeg=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/librsmpeg-74a19ecb8fceb138.rlib --extern serial_test=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libserial_test-c0f974657ac55072.rlib --extern strum_macros=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libstrum_macros-cb0559eb48c6dbe7.so --extern tesseract_sys=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libtesseract_sys-9000effca060dad3.rlib --extern time=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libtime-ca2d80761d324232.rlib --extern url=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/liburl-09e706b4d6247fd9.rlib -Copt-level=3 -Cdebuginfo=2 -Ccodegen-units=1 -Cstrip=none -Cforce-frame-pointers=yes --cap-lints=warn -L native=/usr/lib64 -L native=/usr/lib -L 'native="/usr/lib"' -L native=/usr/lib64 -L native=/usr/lib64 -L 'native="/usr/lib64"' -l avcodec`
error[E0599]: no method named `to_positive_degrees` found for struct `RgbHue<T>` in the current scope
  --> src/hardsubx/imgops.rs:15:22
   |
15 |     *H = hsv_rep.hue.to_positive_degrees();
   |                      ^^^^^^^^^^^^^^^^^^^
   |
help: there is a method `into_positive_degrees` with a similar name
   |
15 |     *H = hsv_rep.hue.into_positive_degrees();
   |                      ++

For more information about this error, try `rustc --explain E0599`.
error: could not compile `ccx_rust` (lib) due to 1 previous error

Caused by:
  process didn't exit successfully: `/usr/bin/rustc --crate-name ccx_rust --edition=2018 src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type staticlib --emit=dep-info,link -C opt-level=3 -C embed-bitcode=no --cfg 'feature="hardsubx_ocr"' --cfg 'feature="leptonica-sys"' --cfg 'feature="rsmpeg"' --cfg 'feature="tesseract-sys"' --check-cfg 'cfg(docsrs,test)' --check-cfg 'cfg(feature, values("enable_ffmpeg", "hardsubx_ocr", "leptonica-sys", "rsmpeg", "sanity_check", "tesseract-sys", "with_libcurl", "wtv_debug"))' -C metadata=52cada234fc33f24 -C extra-filename=-6d45b476bf719ea7 --out-dir /builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps -C strip=debuginfo -L dependency=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps --extern cfg_if=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libcfg_if-9d2dbc57d8792789.rlib --extern clap=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libclap-e5fc0d6cf5f0b011.rlib --extern crc32fast=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libcrc32fast-fbcf64dd4a2f897e.rlib --extern encoding_rs=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libencoding_rs-233b529cd9a1c6e3.rlib --extern env_logger=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libenv_logger-472fef667cb831cb.rlib --extern leptonica_sys=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libleptonica_sys-a9696d6acc70da1d.rlib --extern lib_ccxr=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/liblib_ccxr-4e3e582127ca6239.rlib --extern log=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/liblog-7bc3130406e50940.rlib --extern num_integer=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libnum_integer-69083584aceac81d.rlib --extern palette=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libpalette-b61653662ba8d509.rlib --extern rsmpeg=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/librsmpeg-74a19ecb8fceb138.rlib --extern serial_test=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libserial_test-c0f974657ac55072.rlib --extern strum_macros=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libstrum_macros-cb0559eb48c6dbe7.so --extern tesseract_sys=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libtesseract_sys-9000effca060dad3.rlib --extern time=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/libtime-ca2d80761d324232.rlib --extern url=/builddir/build/BUILD/ccextractor-0.96.4-build/ccextractor/src/rust/../../linux/rust/release/deps/liburl-09e706b4d6247fd9.rlib -Copt-level=3 -Cdebuginfo=2 -Ccodegen-units=1 -Cstrip=none -Cforce-frame-pointers=yes --cap-lints=warn -L native=/usr/lib64 -L native=/usr/lib -L 'native="/usr/lib"' -L native=/usr/lib64 -L native=/usr/lib64 -L 'native="/usr/lib64"' -l avcodec` (exit status: 1)
Failed.
```

Output of `rustc --explain E0599`:
```
This error occurs when a method is used on a type which doesn't implement it:

Erroneous code example:

struct Mouth;

let x = Mouth;
x.chocolate(); // error: no method named `chocolate` found for type `Mouth`
               //        in the current scope

In this case, you need to implement the chocolate method to fix the error:

struct Mouth;

impl Mouth {
    fn chocolate(&self) { // We implement the `chocolate` method here.
        println!("Hmmm! I love chocolate!");
    }
}

let x = Mouth;
x.chocolate(); // ok!
```

--- a/src/rust/build.rs
+++ b/src/rust/build.rs
@@ -84,7 +84,12 @@ fn main() {
     {
         builder = builder.clang_arg("-DENABLE_HARDSUBX");
 
-        // Add FFmpeg include paths for Mac
+        // Check FFMPEG_INCLUDE_DIR environment variable (works on all platforms)
+        if let Ok(ffmpeg_include) = env::var("FFMPEG_INCLUDE_DIR") {
+            builder = builder.clang_arg(format!("-I{}", ffmpeg_include));
+        }
+
+        // Add FFmpeg include paths for Mac (Homebrew)
         if cfg!(target_os = "macos") {
             // Try common Homebrew paths
             if std::path::Path::new("/opt/homebrew/include").exists() {
@@ -110,10 +115,14 @@ fn main() {
                     }
                 }
             }
+        }
 
-            // Also check environment variable
-            if let Ok(ffmpeg_include) = env::var("FFMPEG_INCLUDE_DIR") {
-                builder = builder.clang_arg(format!("-I{}", ffmpeg_include));
+        // On Linux, try pkg-config to find FFmpeg include paths
+        if cfg!(target_os = "linux") {
+            if let Ok(lib) = pkg_config::Config::new().probe("libavcodec") {
+                for path in lib.include_paths {
+                    builder = builder.clang_arg(format!("-I{}", path.display()));
+                }
             }
         }
     }

